File: frontend/src/app/models/productInfo.ts
import {ProductInOrder} from "./ProductInOrder";

export class ProductInfo {
    productId: string;
    productName: string;
    productPrice: number;
    productStock: number;
    productDescription: string;
    productIcon: string;
    productStatus: number; // 0: onsale 1: offsale
    categoryType: number;
    createTime: string;
    updateTime: string;


    constructor(
        productInOrder?: ProductInOrder, 
        productId?: string,
        productName?: string,
        productPrice?: number,
        productStock?: number,
        productDescription?: string,
        productIcon?: string,
        categoryType?: number,
        productStatus?: number
      ) {
        if (productInOrder) {
          this.productId = productInOrder.productId || '';
          this.productName = productInOrder.productName || '';
          this.productPrice = productInOrder.productPrice || 0;
          this.productStock = productInOrder.productStock || 0;
          this.productDescription = productInOrder.productDescription || '';
          this.productIcon = productInOrder.productIcon || '';
          this.categoryType = productInOrder.categoryType || 0;
        } else {
          this.productId = productId; 
          this.productName = productName;
          this.productPrice = productPrice;
          this.productStock = productStock;
          this.productDescription = productDescription;
          this.productIcon = productIcon;
          this.categoryType = categoryType;
          this.productStatus = productStatus;
        }
      }
}

File: backend/src/main/java/com/shop/entity/ProductInfo.java
package com.shop.entity;

import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.Data;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.UpdateTimestamp;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Data
@DynamicUpdate
public class ProductInfo implements Serializable {
    @Id
    private String productId;

    @NotNull
    private String productName;

    @NotNull
    private BigDecimal productPrice;

    @NotNull
    private Integer productSize;

    @NotNull
    @Min(0)
    private Integer productStock;

    private String productDescription;

    private String productIcon;

    @ColumnDefault("0")
    private Integer productStatus;

    @ColumnDefault("0")
    private Integer categoryType;

    @CreationTimestamp
    private Date createTime;
    @UpdateTimestamp
    private Date updateTime;

    public ProductInfo() {
    }
}
